scarlet/lite/display.py

```diff
@@ -0,0 +1,166 @@
+
+    fig, ax = plt.subplots(n_sources, panels, figsize=figsize, squeeze=False)
+
+    marker_kwargs = {"mew": 1, "ms": 10}
+    box_kwargs = {"facecolor": "none", "edgecolor": "w", "lw": 0.5}
+
+    skipped = 0
+    for k, src in enumerate(blend.sources):
+        if src.is_null:
+            skipped += 1
+            continue
+        if use_flux:
+            src_box = src.flux_box
+        else:
+            src_box = src.bbox
+
+        extent = get_extent(src_box)
+
+        # model in its bbox
+        panel = 0
+        model = src.get_model(use_flux=use_flux)
+
+        if show_model:
+            if model_mask:
+                _model_mask = np.max(model, axis=0) <= 0
+            # Show the unrendered model in it's bbox
+            ax[k-skipped][panel].imshow(
+                img_to_rgb(model, norm=norm, channel_map=channel_map, mask=_model_mask),
+                extent=extent,
+                origin="lower",
+            )
+            ax[k-skipped][panel].set_title("Model Source {}".format(k))
+            _add_markers(src, extent, ax[k-skipped][panel], add_markers, False, marker_kwargs, box_kwargs)
+            panel += 1
+
+        # model in observation frame
+        if show_rendered:
+            # Center and show the rendered model
+            model_ = src.get_model(use_flux=use_flux, bbox=bbox)
+            if not use_flux:
+                model_ = observation.render(model_)
+            ax[k-skipped][panel].imshow(
+                img_to_rgb(model_, norm=norm, channel_map=channel_map),
+                extent=get_extent(observation.bbox),
+                origin="lower",
+            )
+            ax[k-skipped][panel].set_title("Model Source {} Rendered".format(k))
+            _add_markers(src, extent, ax[k-skipped][panel], add_markers, add_boxes, marker_kwargs, box_kwargs)
+            panel += 1
+
+        if show_observed:
+            # Center the observation on the source and display it
+            _images = observation.data
+            ax[k-skipped][panel].imshow(
+                img_to_rgb(_images, norm=norm, channel_map=channel_map),
+                extent=get_extent(observation.bbox),
+                origin="lower",
+            )
+            ax[k-skipped][panel].set_title("Observation".format(k))
+            _add_markers(src, extent, ax[k-skipped][panel], add_markers, add_boxes, marker_kwargs, box_kwargs)
+            panel += 1
+
+        if show_spectrum:
+            spectra = [model.sum(axis=(1, 2))]
+
+            for spectrum in spectra:
+                ax[k-skipped][panel].plot(spectrum)
+            ax[k-skipped][panel].set_xticks(range(len(spectrum)))
+            ax[k-skipped][panel].set_title("Spectrum")
+            ax[k-skipped][panel].set_xlabel("Band")
+            ax[k-skipped][panel].set_ylabel("Intensity")
+
+    fig.tight_layout()
+    return fig
+
+
+def compare_spectra(use_flux=True, use_template=True, **all_sources):
```

Comment:
this method is pretty specific: comparing spectra obtained from different runs of the fitter. I'd much rather have this code in user space because I don't think many people would use it. Bit rot is also an issue.